dbms/src/DataStreams/HashJoinProbeBlockInputStream.cpp

```diff
@@ -50,28 +52,29 @@ Block HashJoinProbeBlockInputStream::getTotals()
         }
         join->joinTotals(totals);
     }
+
     return totals;
 }
 
 Block HashJoinProbeBlockInputStream::getHeader() const
 {
     Block res = children.back()->getHeader();
-    join->joinBlock(res);
+    assert(res.rows() == 0);
+    join->joinBlock(res, concurrency_probe_index);
     return res;
 }
 
 Block HashJoinProbeBlockInputStream::readImpl()
 {
-    Block res = children.back()->read();
-    if (!res)
-        return res;
-
-    join->joinBlock(res);
-
-    // TODO split block if block.size() > settings.max_block_size
-    // https://github.com/pingcap/tiflash/issues/3436
+    if (join->needGetNewBlock(concurrency_probe_index))
```

Comment:
Consider if we move `ProbeProcessInfo` to `HashJoinProbeBlockInputStream`, will the code be more simple and clear? `Join` don't need to `initProbeSide`, `concurrency_probe_index` is not needed either.

Reply:
> Consider if we move `ProbeProcessInfo` to `HashJoinProbeBlockInputStream`, will the code be more simple and clear? `Join` don't need to `initProbeSide`, `concurrency_probe_index` is not needed either.

done.